kernel/src/scan/mod.rs

```diff
@@ -105,11 +110,13 @@ pub struct ScanResult {
 
 /// Scan uses this to set up what kinds of columns it is scanning. A reference to the field is
 /// stored so we can get the name and data_type out when building the read expression
-enum ColumnType<'a> {
-    Selected(&'a StructField),
-    Partition(&'a StructField),
+pub enum ColumnType<'a> {
+    Selected(Cow<'a, StructField>),
+    Partition(Cow<'a, StructField>),
```

Comment:
What is the use case for introducing `Cow`, out of curiosity?

Reply:
This was left from when I wanted a global scan state that included these, and I wanted to become owned so it could be passed around. That wasn't a great idea so I removed it, but didn't clean this up. Reverting this change now.